Presentation/PresentationOptions.swift

```diff
@@ -99,6 +103,14 @@ public extension UIViewController {
     }
 }
 
+internal extension PresentationOptions {
+    /// Will hide the navigation bar
+    static let prefersNavigationBarHidden = PresentationOptions()
+
+    /// Will show the navigation bar
+    static let prefersNavigationBarShown = PresentationOptions()
```

Comment:
Should not this be the default? do we need this option?
It makes it possible to call `present(vc, options: [. prefersNavigationBarHidden, . prefersNavigationBarShown])`
What should be the behavior in this case?

Reply:
These options are in the internal extension of PresentationOptions. So you cannot specify them directly together like this. you have to use a static function instead, which is kind of intuitive that you should only use one. But if you workaround it like 
`present(vc, options: [PresentationOptions. prefersNavigationBarHidden(true), PresentationOptions. prefersNavigationBarHidden(true)])`

It will consider the hidden behaviour as priority. We are considering logging the ambiguity here.

And to answer, "Do we need this option?" I would say, this is our attempt to push presentations into navigation stack, where some prefer it hidden, and some don't.

We would love to know if there is any other approach we should take to achieve this.

Reply:
Also this is supposed to add only preference for the Presentation. If unspecified, we would want the existing navigationController specifications to be kept. This is not to introduce a breaking change where the NavigationBar appearance is setup after the instantiation of the UINavigationController

Reply:
Ok I understand. Thanks for the clarification.

Is this only applied when pushing to a navigationController? Should it also work when modally presented?

Reply:
It works even if modally presented.

Reply:
so if we do:
```swift
present(vc, style: .modal, options: [.embedInNavigationController, .prefersNavigationBarHidden(true)])
```
I think that the navigation bar will be visible, since it's not calling present of UINavigationController

Reply:
Thanks! FIxed it now.